packages/fxa-content-server/app/scripts/views/mixins/email-autocomplete-domains-mixin.js

```diff
@@ -0,0 +1,124 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * A mixin to autocomplete common domains from `datalist option` `value`s.
+ */
+import Template from 'templates/partial/email-autocomplete-domains.mustache';
+import UserAgentMixin from '../../lib/user-agent-mixin';
+
+export const EMAIL_SELECTOR = 'input[type=email]';
+export const DATALIST_OPTIONS_SELECTOR = '#autocomplete-domain option';
+export const FOCUS_HACK_SELECTOR = '#focus-hack';
+
+export const DOMAINS = [
+  'gmail.com',
+  'outlook.com',
+  'hotmail.com',
+  'yahoo.com',
+  'qq.com',
+  'web.de',
+  'aol.com',
+  'mail.ru',
+  'icloud.com',
+  'gmx.de',
+  't-online.de',
+  'orange.fr',
+  'yandex.ru',
+  'yahoo.fr',
+  'live.com',
+  '163.com',
+  'msn.com',
+  'comcast.net',
+  'hotmail.co.uk',
+  'hotmail.fr',
+];
+
+export default {
+  events: {
+    [`keyup ${EMAIL_SELECTOR}`]: '_toggleDomainAutocomplete',
+  },
+  dependsOn: [UserAgentMixin],
+
+  setInitialContext(context) {
+    const isAndroid = this.getUserAgent().isAndroid();
+    const isChrome = this.getUserAgent().isChrome();
+
+    const unsafeEmailAutocompleteDomainsHTML = this.renderTemplate(Template, {
+      options: DOMAINS,
+    });
+    context.set({
+      unsafeEmailAutocompleteDomainsHTML,
+      isAndroid,
+      isChrome,
+    });
+  },
+
+  afterRender() {
+    this.input = this.$(EMAIL_SELECTOR);
+    this.datalistOptions = this.$(DATALIST_OPTIONS_SELECTOR);
+    this.focusHack = this.$(FOCUS_HACK_SELECTOR);
+    this.previousUsername = null;
+  },
+
+  /**
+   * A catch-most-browsers hack when datalist options are modified.
+   * Chrome otherwise shows all options when '@' is typed,
+   * Safari immediately goes to the dropdown preventing users
+   * from typing further, and FF will not update the dropdown on `keyup`
+   * and has a known bug with dynamic datalists
+   * https://bugzilla.mozilla.org/show_bug.cgi?id=1474137
+   */
+  _focusHack() {
+    this.focusHack.focus();
```

Comment:
So focus an element after the datalist then focus the input element again. Will screen readers be happy with this?

Reply:
I actually tested this in VoiceOver on Mac on FireFox to make sure it didn't cause any mishaps and it doesn't seem to complain. Granted I didn't test VO on other browsers. 😕